### DISCUSSION TITLE
Custom modules won't reload without scheduler restart

### DISCUSSION THREAD

**Discussion by Travior** [apache/airflow#31932] [Q&A]
Hi everyone,
I'm currently facing the annoying issue that custom modules won't reload without restarting the scheduler.
I know that this behavior is intended for triggers, however I'm unsure whether this is intended for modules that get imported in DAGs as well.

The file structure is as follows:
```
|- dags
|   |- mydag.py
|   |- utils
|   |   |- __init__.py
|   |   |- old_module.py
|   |   |- new_module.py
``` 
I can currently import `old_module.py` in `mydag.py` without any issue, however I get an import error when importing `new_module.py` in my DAGs. Additionally changes to `old_module.py` won't get picked up. In the past I was able to resolve this with a scheduler restart, this is however not really feasable for every change to the modules.

I don't think that this is a path issue, since I'm able to import `old_module.py` just fine, but I'm running out of ideas on what the issue could be.

Could this issue have something to do with the fact that I don't have an .airflowignore file present? If I were to exclude the utils folder with such a file would that fix the issue?

**Comment by sqrs** [comment]
in your dag file,
try reload method with importlib

ex)
```
from config import config
from common import models

import importlib
importlib.reload(config)
importlib.reload(models)
```

**Comment by potiuk** [comment]
Airflow will reparse DAGs periodically every https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#min-file-process-interval and then it should load all the modules available. Maybe your interval is too long for you and you are impatient waiting - shortening the time will make airflow re-parse DAG more often at the expense of more CPU used for parsing. There are also various strategies you can do to speed up parsing and prioritise DAGs that changed recently - see at all the parsing-related parameters in configuration.
